server/tests/test_output_redaction.py

```
@@ -0,0 +1,148 @@
+class TestTwoHookIntegration:
+    """Integration across Hook 1 (tool_completion) and Hook 2 (db_save).
+
+    Proves that a secret produced by a tool is redacted exactly once on the
+    primary path and that the secondary hook is a true no-op in steady state,
+    matching the belt-and-suspenders invariant.
+    """
+
+    def test_hook2_is_noop_on_output_already_redacted_by_hook1(self):
+        raw = "AWS_ACCESS_KEY_ID=AKIA6ODU7H4ZLXKDNQ3X"
+        hook1_out, hook1_findings = redact(raw)
+        assert hook1_findings and "[REDACTED:" in hook1_out
```

Ruff (0.15.12)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)